maintainers/scripts/haskell/update-stackage.sh

```diff
@@ -0,0 +1,51 @@
+#! /usr/bin/env bash
+
+set -eu -o pipefail
+
+tmpfile=$(mktemp "update-stackage.XXXXXXX")
+# shellcheck disable=SC2064
+trap "rm ${tmpfile} ${tmpfile}.new" 0
+
+curl -L -s "https://stackage.org/nightly/cabal.config" >"$tmpfile"
+version=$(sed -rn "s/^--.*http:..(www.)?stackage.org.snapshot.nightly-//p" "$tmpfile")
+
+# Create a simple yaml version of the file.
+sed -r \
+    -e '/^--/d' \
+    -e 's|^constraints:||' \
+    -e 's|^ +|  - |' \
+    -e 's|,$||' \
+    -e '/installed$/d' \
+    -e '/^$/d' \
+    < "${tmpfile}" | sort --ignore-case >"${tmpfile}.new"
+
+# Drop restrictions on some tools where we always want the latest version.
+sed -r \
+    -e '/ cabal-install /d' \
+    -e '/ cabal2nix /d' \
+    -e '/ cabal2spec /d' \
+    -e '/ distribution-nixpkgs /d' \
+    -e '/ git-annex /d' \
+    -e '/ hindent /d' \
+    -e '/ hledger/d' \
+    -e '/ hlint /d' \
+    -e '/ hoogle /d' \
+    -e '/ hopenssl /d' \
+    -e '/ jailbreak-cabal /d' \
+    -e '/ json-autotype/d' \
+    -e '/ language-nix /d' \
+    -e '/ shake /d' \
+    -e '/ ShellCheck /d' \
+    -e '/ stack /d' \
+    -e '/ weeder /d' \
+    < "${tmpfile}.new" > "${tmpfile}"
```

Comment:
Now that we're on Stackage Nightly, I imagine we can drop quite a few of these lines here, since these packages will generally also be the latest version on Hackage.  That is to say, the version of the package in Stackage Nightly should also be the latest version of the package on Hackage.

If the packages in Stackage Nightly are _not_ the latest version on Hackage, then the Stackage maintainers have probably done that for a good reason.  For instance, the packages aren't compatible and don't compile together.

Although this is not something that should block this PR.  We can do this at some point in  the future.

Reply:
We should drop every one of these except the ones that are nixpkgs-related `jailbreak-cabal`, `cabal2nix`, ...

Reply:
Yeah, let’s clean that up. But that requires a bit of careful deliberation to see if we break anything right now, so I would like this to happen in an independent PR.